frontend/public/components/edit-yaml.jsx

```diff
@@ -246,6 +248,14 @@ export const EditYAML = connect(stateToProps)(
         obj.metadata.namespace = this.props.activeNamespace;
       }
 
+      if (this.props.activePerspective === 'dev') {
+        if (obj.metadata.labels && !obj.metadata.labels['app.kubernetes.io/part-of']) {
```

Comment:
This could be one line:
```suggestion
        obj.metadata.labels = { ...obj.metadata.labels, 'app.kubernetes.io/part-of': this.props.activeApplication };
```

Reply:
Yes, I also started with this but what if the user adds the label manually or the file user imports already has label in it, do we want to override it with the `activeApplication` ?

Reply:
@divyanshiGupta oops i mixed up the order in my one liner :P
```suggestion
        obj.metadata.labels = { 'app.kubernetes.io/part-of': this.props.activeApplication, ...obj.metadata.labels };
```

Reply:
No, right now `all applications` will always be there even if the user has not selected an application. But since that is not a valid application it will show error msg. So `activeApplication` will never e null/undefined

Reply:
done

Reply:
@divyanshiGupta  hmmm i didn't realize that activeApplication will do that. It should probably be null / undefined if no application is active.

Reply:
Yes we don't assign it in the label, it shows an error that it is not a valid name. And we also do not have a check for that. So for the first time when there is no application selected we want to show select an application ?

Reply:
fix the application selector to store an undefined value for the active application if none selected.

Reply:
Done. But we still want to reset it to all apps when the user switches namespace, right?

Reply:
The behavior should remain the same but now I get this when I refresh the page:
![image](https://user-images.githubusercontent.com/14068621/57954480-09cd9500-78c1-11e9-865d-55cc281c3137.png)

Reply:
So I have a few questions now 
- What should be the value of title and activeApplication when user hits the root URL? In namespace dropdown, title and activeNamespace gets set as the lastViewedNamespace/default.
Currently in this case I am setting the title and activeApplication as all applications because that is  the default value because we decided to show all applications when user switches namespaces.
- What should be the value of title and activeApplication when user refreshes the page? In namespace dropdown, title and activeNamespace gets set as the namespace present in the URL, if there is no namespace in the URL it gets set as the lastViewedNamespace. Currently in this case also it gets set to the default value that is all applications because we don't have namespace in the URL and we are not saving lastViewedApplication in localStorage as well.

Reply:
I have made the following changes
- For the first when there is no application selected, select an application will be shown and there will be no active-app
- When the user refreshes the page, app-switcher will show the last-viewed-app and that will be set as the active-app
- When the user switches namespace, all-app will be shown and set as active-app and last-viewed-app

Since there is no UX for this and a lot of edge cases, we cant go over all edge cases right now. If UX feels something is missing they can always raise bugs or as we talked previously we will have a separate story to cover other edge cases.

Reply:
I don't know, having `#ALL_APP#` as a key when user first comes to the page seems okay to me. This is how console handles all projects as well. They have this key `#ALL_NS#` added to the store which also acts like a validator for wrong input value. 
Now the flow seems inconsistent to me -
- When user comes to this page he has to select an application.
- But as soon as he selects a namespace he has all applications selected for him which errors out when creating the resource?
- And then there's a last viewed application added again? I thought we weren't going to store that?

I would rather have all application selected for the user when he comes to any page or when he changes namespace or refreshes the page. Having same flow in every scenarios makes it simpler for the user to understand.

Reply:
Also, having all this changes in behaviour of application switcher and its actions and reducers seems out of scope for this story. We should probably have a separate story for changing use cases for application switcher after we've thoroughly vetted all the use cases and edge case scenarios with UX/PM.

Reply:
I think it's better to get this in without any changes to application switcher. We can make any changes required to application switcher in a separate story. @christianvogt what do you think?

Reply:
that's fine, we can address other issues w/ bugs. Still waiting on confirmation on applying the labels in general.